hollow/src/main/java/com/netflix/hollow/api/producer/validation/ObjectModificationValidator.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.producer.validation;
+
+import com.netflix.hollow.api.custom.HollowAPI;
+import com.netflix.hollow.api.objects.HollowObject;
+import com.netflix.hollow.api.producer.HollowProducer.ReadState;
+import com.netflix.hollow.api.producer.HollowProducer.Validator;
+import com.netflix.hollow.core.HollowConstants;
+import com.netflix.hollow.core.index.HollowPrimaryKeyIndex;
+import com.netflix.hollow.core.index.key.PrimaryKey;
+import com.netflix.hollow.core.read.dataaccess.HollowDataAccess;
+import com.netflix.hollow.core.read.engine.HollowTypeReadState;
+import com.netflix.hollow.core.read.engine.object.HollowObjectTypeReadState;
+import com.netflix.hollow.core.schema.HollowSchema.SchemaType;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+import java.util.function.Function;
+
+/**
+ * This validator runs a provided {@link BiPredicate} on every instance of an object that was
+ * modified in a cycle. This "filter" function is called with the old object as the first argument
+ * and the new object as the second argument. If the function returns false, this validator is
+ * marked as failed and the producer cycle fails.
+ * Note that the filter function is not called with any objects that were added or removed.
+ * Also note that this validator can only be used on types that have a primary key.
+ * Example construction - note that `Movie` and `MovieAPI are generated classes:
```

Comment:
missing a closing backtick here - adding it now

Reply:
Strictly speaking `{@code Movie}` and `{@code MovieAPI}` is what you want.

I wonder if there's such a thing as a javadoc postprocessor that adds Markdown support?

Reply:
Fair point, updated